src/updateState.js

```diff
@@ -3,14 +3,16 @@
  * The only way to change the data in the store is to call `dispatch()` on it.
  * @param { Object } state The initial application state
  * @return {Function} currentState Return a function that
- * update and return the current state
+ * updates and returns the current state
  */
-export default ( state ) => {
-  return ( function( state ) {
-    var state = Object.assign({}, state)
-    return current => {
-      state = Object.assign({}, state, current)
-      return state
-    }
-  }(state))
+export default (state) => {
+  return (
+    function(state) {
+      var state = Object.assign({}, state)
```

Comment:
If object spread is enabled this could be rewritten as:

```javascript
let state = { ...state };
return current => {
  state = { ...state, ...current }
}
```
I believe.